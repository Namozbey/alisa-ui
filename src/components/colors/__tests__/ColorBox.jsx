import React from 'react'
import { shallow } from 'enzyme'

import { ColorBox } from '..'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)
  const { color, hex, children } = props
  const wrapper = shallow(
    <ColorBox color={color} hex={hex}>
      {children}
    </ColorBox>,
  )

  return {
    props,
    wrapper,
  }
}

describe('ColorBox', () => {
  it('renders matching snapshot for color level 50', () => {
    const { wrapper } = setup({ color: 'teal-50' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 100', () => {
    const { wrapper } = setup({ color: 'teal-100' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 200', () => {
    const { wrapper } = setup({ color: 'teal-200' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 300', () => {
    const { wrapper } = setup({ color: 'teal-300' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 400', () => {
    const { wrapper } = setup({ color: 'teal-400' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 500', () => {
    const { wrapper } = setup({ color: 'teal-500' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 600', () => {
    const { wrapper } = setup({ color: 'teal-600' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 700', () => {
    const { wrapper } = setup({ color: 'teal-700' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 800', () => {
    const { wrapper } = setup({ color: 'teal-800' })

    expect(wrapper).toMatchSnapshot()
  })

  it('renders matching snapshot for color level 900', () => {
    const { wrapper } = setup({ color: 'teal-900' })

    expect(wrapper).toMatchSnapshot()
  })
})
