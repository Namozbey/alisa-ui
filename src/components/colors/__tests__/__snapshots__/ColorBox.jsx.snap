// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorBox renders matching snapshot for color level 50 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-50"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-50
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #f1fdfd
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 100 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-100"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-100
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #e3fbfb
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 200 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-200"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-200
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #bbf6f4
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 300 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-300"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-300
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #8bede5
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 400 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-400"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-400
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #62e4d2
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 500 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-500"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-500
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #37d9b8
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 600 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-600"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-600
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #1ac09c
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 700 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-700"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-700
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #0d9676
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 800 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-800"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-800
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #046b53
    </div>
  </div>
</div>
`;

exports[`ColorBox renders matching snapshot for color level 900 1`] = `
<div
  style={
    Object {
      "width": 150,
    }
  }
>
  <div
    className="bg-teal-900"
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 5,
        "color": "#000",
        "display": "flex",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  />
  <div
    className="px-0.5 md:flex md:justify-between md:space-x-2 2xl:space-x-0 2xl:block"
  >
    <div
      className="text-xs font-medium text-gray-900"
    >
      teal-900
    </div>
    <div
      className="text-xs text-gray-500"
    >
      #003833
    </div>
  </div>
</div>
`;
